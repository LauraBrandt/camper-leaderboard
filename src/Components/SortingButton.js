import React from 'react';
import PropTypes from 'prop-types';

const SortingButton = ({ type, direction, inactive, handleClick }) => {
  return(
    <div>
      <button 
        className={`${type} ${direction}${inactive ? ' inactive' : ''}`} 
        title={`Sort ${direction === 'up' ? 'descending' : 'ascending'}`}
        onClick={inactive ? null : handleClick}
      >
        <i className={`fas fa-caret-${direction}`}></i>
      </button>
    </div>
  );
}

SortingButton.propTypes = {
  type: PropTypes.string.isRequired,
  direction: PropTypes.oneOf(['up', 'down']),
  handleClick: PropTypes.func.isRequired,
  inactive: PropTypes.bool
}

export default SortingButton;
