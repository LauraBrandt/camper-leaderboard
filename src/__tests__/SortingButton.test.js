import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import SortingButton from '../Components/SortingButton';

describe('SortingButton', () => {
  let mockProps = {
    type: "recent",
    direction: "up",
    handleClick: jest.fn()
  };

  const resetProps = () => mockProps = {
    type: "recent",
    direction: "up",
    handleClick: jest.fn()
  }
  
  it('renders and matches snapshot', () => {
    const renderedComponent = renderer.create(<SortingButton {...mockProps}/>);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot of new props', () => {
    mockProps.type = "alltime";
    mockProps.direction = "down";
    const renderedComponent = renderer.create(<SortingButton {...mockProps}/>);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  describe('content', () => {
    let component;

    beforeEach(() => {
      component = shallow(<SortingButton {...mockProps}/>);
    });
    
    afterEach(() => {
      resetProps();
    });

    it('contains a button', () => {
      expect(component.find('button')).toHaveLength(1);
    });

    it('has the correct icon', () => {
      expect(component.find('.fa-caret-up')).toHaveLength(1);
      
      component.setProps({ direction: 'down' });
      expect(component.find('.fa-caret-down')).toHaveLength(1);
    });

    it('has the correct button classes', () => {
      expect(component.find('button').prop('className')).toBe('recent up');
      
      component.setProps({inactive: true});
      expect(component.find('button').prop('className')).toBe('recent up inactive');
    });

    it('has the correct button title', () => {
      expect(component.find('button').prop('title')).toBe('Sort descending');
      
      component.setProps({ direction: 'down' });
      expect(component.find('button').prop('title')).toBe('Sort ascending');
    });
  });

  describe('interactions', () => {
    let component;
    beforeEach(() => {
      component = shallow(<SortingButton {...mockProps}/>);
    });
    
    afterEach(() => {
      resetProps();
    });

    it('calls handleClick on button click', () => {
      expect(mockProps.handleClick).not.toBeCalled();

      component.find('button').simulate('click');
      expect(mockProps.handleClick).toBeCalled();
    });

    it("doesn't call handleClick when inactive", () => {
      component.setProps({ inactive: true });

      expect(mockProps.handleClick).not.toBeCalled();

      component.find('button').simulate('click');
      expect(mockProps.handleClick).not.toBeCalled();
    });
  });
});