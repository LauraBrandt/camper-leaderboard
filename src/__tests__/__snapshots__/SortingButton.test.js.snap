// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingButton matches snapshot of new props 1`] = `
<div>
  <button
    className="alltime down"
    onClick={[Function]}
    title="Sort ascending"
  >
    <i
      className="fas fa-caret-down"
    />
  </button>
</div>
`;

exports[`SortingButton renders and matches snapshot 1`] = `
<div>
  <button
    className="recent up"
    onClick={[Function]}
    title="Sort descending"
  >
    <i
      className="fas fa-caret-up"
    />
  </button>
</div>
`;
